sky/check.py

```diff
@@ -336,10 +339,54 @@ def get_cloud_credential_file_mounts(
     return file_mounts
 
 
-def _format_enabled_cloud(cloud_name: str) -> str:
+def _print_checked_cloud(
+    echo: Callable, verbose: bool, cloud_tuple: Tuple[str,
+                                                      Union[sky_clouds.Cloud,
+                                                            ModuleType]],
+    cloud_capabilities: List[Tuple[CloudCapability, bool,
+                                   Optional[str]]]) -> None:
+    cloud_repr, cloud = cloud_tuple
+    # Print the capabilities for the cloud.
+    # consider cloud enabled if any capability is enabled.
+    enabled_capabilities: List[CloudCapability] = []
+    hints_to_capabilities: Dict[str, List[CloudCapability]] = {}
+    reasons_to_capabilities: Dict[str, List[CloudCapability]] = {}
+    for capability, ok, reason in cloud_capabilities:
+        if ok:
+            enabled_capabilities.append(capability)
+            if reason is not None:
+                hints_to_capabilities.setdefault(reason, []).append(capability)
+        elif reason is not None:
+            reasons_to_capabilities.setdefault(reason, []).append(capability)
+    status_msg = 'enabled' if enabled_capabilities else 'disabled'
+    styles = {
+        'fg': 'green',
+        'bold': False
+    } if enabled_capabilities else {
+        'dim': True
+    }
+    service_string = (click.style('[' + ', '.join(enabled_capabilities) + ']',
+                                  **styles) if enabled_capabilities else '')
+    echo('  ' + click.style(f'{cloud_repr}: {status_msg}', **styles) + ' ' +
+         service_string)
+    if status_msg == 'enabled':
+        if verbose and cloud is not cloudflare:
+            activated_account = cloud.get_active_user_identity_str()
+            if activated_account is not None:
+                echo(f'    Activated account: {activated_account}')
+    for reason, caps in hints_to_capabilities.items():
+        echo(f'    Hint [{", ".join(caps)}]: {reason}')
+    for reason, caps in reasons_to_capabilities.items():
+        echo(f'    Reason [{", ".join(caps)}]: {reason}')
+
+
+def _format_enabled_cloud(cloud_name: str,
+                          capabilities: List[CloudCapability]) -> str:
 
     def _green_color(cloud_name: str) -> str:
-        return f'{colorama.Fore.GREEN}{cloud_name}{colorama.Style.RESET_ALL}'
+        enabled_capabilities_str = '[' + ', '.join(capabilities) + ']'
```

Comment:
This `_green_color` function is just designed to make some text green. We shouldn't add the logic here - it should be added wherever `_green_color` is called.